src/Components/Components/src/PublicAPI.Unshipped.txt

```diff
@@ -1,6 +1,20 @@
 #nullable enable
 *REMOVED*static Microsoft.AspNetCore.Components.ParameterView.FromDictionary(System.Collections.Generic.IDictionary<string!, object!>! parameters) -> Microsoft.AspNetCore.Components.ParameterView
 *REMOVED*virtual Microsoft.AspNetCore.Components.RenderTree.Renderer.DispatchEventAsync(ulong eventHandlerId, Microsoft.AspNetCore.Components.RenderTree.EventFieldInfo! fieldInfo, System.EventArgs! eventArgs) -> System.Threading.Tasks.Task!
+Microsoft.AspNetCore.Components.ComponentApplicationState
+Microsoft.AspNetCore.Components.ComponentApplicationState.PersistAsJson<T>(string! key, T instance) -> void
+Microsoft.AspNetCore.Components.ComponentApplicationState.PersistState(string! key, byte[]! value) -> void
+Microsoft.AspNetCore.Components.ComponentApplicationState.RegisterOnPersistingCallback(System.Func<System.Threading.Tasks.Task!>! callback) -> void
```

Comment:
Should there be a corresponding `RemoveOnPersistingCallback`? Otherwise it seems like people would be forced to leak memory if they are doing this from some component or service that has a lifetime shorter than the application itself.

Reply:
That's a good suggestion, I'll add it

Reply:
I've switched this to be a full fledged event. I didn't like the name `OnPersisting` (which is what I've chosen so far) so I'm extremely open to alternative names. I want to avoid generic terms like PersistingApplicationState and similar, since I find them too generic and likely overloaded